contracts/orderValidation/OrderValidatorV1.sol

```diff
@@ -0,0 +1,439 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+// OZ dependencies
+import {IERC1271} from "@openzeppelin/contracts/interfaces/IERC1271.sol";
+import {IERC2981} from "@openzeppelin/contracts/interfaces/IERC2981.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {IERC165, IERC721} from "@openzeppelin/contracts/token/ERC721/IERC721.sol";
+import {IERC1155} from "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+
+// LooksRare libraries and validation code constants
+import {OrderTypes} from "../libraries/OrderTypes.sol";
+import "./ValidationCodeConstants.sol";
+
+// LooksRare interfaces
+import {ICurrencyManager} from "../interfaces/ICurrencyManager.sol";
+import {IExecutionManager} from "../interfaces/IExecutionManager.sol";
+import {IExecutionStrategy} from "../interfaces/IExecutionStrategy.sol";
+import {IRoyaltyFeeRegistry} from "../interfaces/IRoyaltyFeeRegistry.sol";
+import {ITransferManagerNFT} from "../interfaces/ITransferManagerNFT.sol";
+import {ITransferSelectorNFTExtended, IRoyaltyFeeManagerExtended} from "./ExtendedInterfaces.sol";
+
+// LooksRareExchange
+import {LooksRareExchange} from "../LooksRareExchange.sol";
+
+/**
+ * @title OrderValidatorV1
+ */
+contract OrderValidatorV1 {
+    using OrderTypes for OrderTypes.MakerOrder;
+
+    // ERC721 interfaceID
+    bytes4 public constant INTERFACE_ID_ERC721 = 0x80ac58cd;
+
+    // ERC1155 interfaceID
+    bytes4 public constant INTERFACE_ID_ERC1155 = 0xd9b67a26;
+
+    // TransferManager ERC721
+    address public immutable TRANSFER_MANAGER_ERC721;
+
+    // TransferManager ERC1155
+    address public immutable TRANSFER_MANAGER_ERC1155;
+
+    // Domain separator from LooksRare Exchange
+    bytes32 public immutable _DOMAIN_SEPARATOR;
+
+    // LooksRare Exchange
+    LooksRareExchange public immutable looksRareExchange;
+
+    // Currency Manager
+    ICurrencyManager public currencyManager;
+
+    // Execution Manager
+    IExecutionManager public executionManager;
+
+    // Transfer Selector
+    ITransferSelectorNFTExtended public transferSelectorNFT;
+
+    // Royalty Fee Registry
+    IRoyaltyFeeRegistry public royaltyFeeRegistry;
+
+    /**
+     * @notice Constructor
+     * @param _looksRareExchange address of the LooksRare exchange (v1)
+     */
+    constructor(address _looksRareExchange) {
+        looksRareExchange = LooksRareExchange(_looksRareExchange);
+        _DOMAIN_SEPARATOR = LooksRareExchange(_looksRareExchange).DOMAIN_SEPARATOR();
+
+        TRANSFER_MANAGER_ERC721 = ITransferSelectorNFTExtended(
+            address(LooksRareExchange(_looksRareExchange).transferSelectorNFT())
```

Comment:
Can you make `address(LooksRareExchange(_looksRareExchange).transferSelectorNFT())` so it is not called twice?

Reply:
Also, since we have it already, why don't we assign it to `transferSelectorNFT` too (it's currently only assigned in `updatePeripheralContractAddresses`)?

Reply:
I have moved the entire logic in the constructor instead since these are likely not to be updated.